core/src/main/scalanative/sttp/client4/curl/AbstractCurlBackend.scala

```diff
@@ -62,6 +86,8 @@ abstract class AbstractCurlBackend[F[_]](_monad: MonadError[F], verbose: Boolean
           case None       => handleBase(request, curl, spaces)
         }
       }
+
+      Context(ctx => perform(ctx))
```

Comment:
at first sight I thought that this creates a new context, but I think it runs the given operation using a context in safe way? Maybe instead of `apply`, we could have a `evaluateUsing` method or sth like that